pkg/volume/csi/csi_attacher.go

```diff
@@ -315,7 +298,7 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		if err != nil && volumetypes.IsOperationFinishedError(err) {
 			// clean up metadata
 			klog.Errorf(log("attacher.MountDevice failed: %v", err))
-			if err := removeMountDir(c.plugin, deviceMountPath); err != nil {
+			if err := removeMountDir(c.plugin, dataDir); err != nil {
```

Comment:
Again, whole deviceMountPath should be removed.

Reply:
this needs to be addressed still..

Reply:
This was fixed.